packages/core-data/src/private-actions.js

```diff
@@ -0,0 +1,16 @@
+/**
```

Comment:
I think we need to import this file in the index and register the private actions [here](https://github.com/WordPress/gutenberg/blob/08d58182caf34f77efc39c303e59c7352c6fbb78/packages/core-data/src/index.js#L81):

```
unlock( store ).registerPrivateActions( privateActions );
```

Reply:
I was running a problem with the logic not working as expected while experimenting with the PR locally, and I can confirm that this fixes the issue 👍